source/slang/ir-legalize-types.cpp

```diff
@@ -388,6 +412,42 @@ static LegalVal legalizeOperand(
     return LegalVal::simple(irValue);
 }
 
+static void getArgumentValues(
+    List<IRValue*> & instArgs, 
+    LegalVal val)
+{
+    switch (val.flavor)
+    {
+    case LegalVal::Flavor::simple:
+        instArgs.Add(val.getSimple());
+        break;
+    case LegalVal::Flavor::implicitDeref:
+        getArgumentValues(instArgs, val.getImplicitDeref());
+        break;
+    case LegalVal::Flavor::tuple:
+        {
+            for (auto elem : val.getTuple()->elements)
+                getArgumentValues(instArgs, elem.val);
+        }
+        break;
+    }
+}
+
+static LegalVal legalizeCall(
+    TypeLegalizationContext*    context,
+    IRCall* callInst)
+{
+    // TODO: implement legalization of non-simple return types
+    auto retType = legalizeType(context, callInst->type);
+    SLANG_ASSERT(retType.flavor == LegalType::Flavor::simple);
+    
```

Comment:
Handling the `implicitDeref` case here would seem simple enough, but we can get to that when it comes up.